pkg/collector/corechecks/cluster/orchestrator/processors/processor_test.go

```diff
@@ -18,28 +18,103 @@ type Item struct {
 	UID string
 }
 
-func TestChunkItems(t *testing.T) {
-	items := []interface{}{
+func TestChunkOrchestratorMetadataBySizeAndWeight(t *testing.T) {
+	orchestratorMetadata := []interface{}{
 		Item{UID: "1"},
 		Item{UID: "2"},
 		Item{UID: "3"},
 		Item{UID: "4"},
 		Item{UID: "5"},
 	}
-	expected := [][]interface{}{
+	tests := []struct {
+		name                     string
+		maxChunkSize             int
+		maxChunkWeight           int
+		orchestratorMetadata     []interface{}
+		orchestratorMetadataYaml [][]byte
+		expectedChunks           [][]interface{}
+	}{
 		{
-			Item{UID: "1"},
-			Item{UID: "2"},
+			name:                 "chunk by size and weight, one high weight",
+			maxChunkSize:         3,
+			maxChunkWeight:       1000,
+			orchestratorMetadata: orchestratorMetadata,
+			orchestratorMetadataYaml: [][]byte{
+				make([]byte, 1001),
+				make([]byte, 100),
+				make([]byte, 100),
+				make([]byte, 100),
+				make([]byte, 100),
+			},
+			expectedChunks: [][]interface{}{
+				{Item{UID: "1"}},
+				{Item{UID: "2"}, Item{UID: "3"}, Item{UID: "4"}},
+				{Item{UID: "5"}},
+			},
 		},
 		{
-			Item{UID: "3"},
-			Item{UID: "4"},
+			name:                 "chunk by size and weight, weight exceeded",
+			maxChunkSize:         3,
+			maxChunkWeight:       1000,
+			orchestratorMetadata: orchestratorMetadata,
+			orchestratorMetadataYaml: [][]byte{
+				make([]byte, 2000),
+				make([]byte, 2000),
+				make([]byte, 2000),
+				make([]byte, 2000),
+				make([]byte, 2000),
+			},
+			expectedChunks: [][]interface{}{
+				{Item{UID: "1"}},
+				{Item{UID: "2"}},
+				{Item{UID: "3"}},
+				{Item{UID: "4"}},
+				{Item{UID: "5"}},
+			},
 		},
 		{
-			Item{UID: "5"},
+			name:                 "chunk by size and weight, low weight",
+			maxChunkSize:         3,
+			maxChunkWeight:       1000,
+			orchestratorMetadata: orchestratorMetadata,
+			orchestratorMetadataYaml: [][]byte{
+				make([]byte, 100),
+				make([]byte, 100),
+				make([]byte, 100),
+				make([]byte, 100),
+				make([]byte, 100),
+			},
+			expectedChunks: [][]interface{}{
+				{Item{UID: "1"}, Item{UID: "2"}, Item{UID: "3"}},
+				{Item{UID: "4"}, Item{UID: "5"}},
+			},
+		},
+		{
+			name:                 "chunk by size and weight, mixed",
```

Comment:
can we add a test where we have something like [500,300,200,500]
I assume the chunk would be the following:
- [500,300,200]
- [500]
can you confirm? Does it include 1k itself?

Reply:
It includes the limit, I added one test `"chunk by size and weight, include limit itself"`